samples/Particles/ParticlesSample/ParticlesSample.Wpf/Win32GameControl.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Interop;
+using System.Windows.Media;
+using Stride.Core.Presentation.Controls;
+using Stride.Engine;
+using Stride.Games;
+
+namespace ParticlesSample
+{
+    internal class Win32GameControl : FrameworkElement
+    {
+        public Win32GameControl()
+        {
+            Loaded += Win32GameHost_Loaded;
+            Unloaded += Win32GameHost_Unloaded;
+        }
+
+        protected override int VisualChildrenCount => _gameEngineHost is null ? 0 : 1;
+
+        protected override Visual GetVisualChild(int index)
+        {
+            if (index == 0 && _gameEngineHost is not null) return _gameEngineHost;
+            throw new ArgumentOutOfRangeException(nameof(index));
+        }
+
+        protected override Size ArrangeOverride(Size finalSize)
+        {
+            _gameEngineHost?.Arrange(new Rect(finalSize));
+            return finalSize;
+        }
+
+        protected override Size MeasureOverride(Size availableSize)
+        {
+            if (_gameEngineHost is not null)
+            {
+                _gameEngineHost.Measure(availableSize);
+                return _gameEngineHost.DesiredSize;
+            }
+
+            return availableSize;
+        }
+
+        private void Win32GameHost_Loaded(object sender, RoutedEventArgs e)
+        {
+            // Get the handle for the content window
+            var host = Window.GetWindow(this);
+            if (host is null) return;
+            var wih = new WindowInteropHelper(host);
+            var hostHandle = wih.Handle;
+
+            // Create a child window in which to host the game
+            var className = GetType().Name;
+            var wndClass = new NativeMethods.WndClassEx();
+            wndClass.cbSize = (uint)Marshal.SizeOf(wndClass);
+            wndClass.hInstance = NativeMethods.GetModuleHandle(null);
+            wndClass.lpfnWndProc = NativeMethods.DefaultWindowProc;
```

Comment:
I think this causes the problems with resizing and dragging. It shouldn't be needed to run the rendering in a task, either.

Take a look at [this](https://github.com/terrafx/terrafx/blob/3e3fa918543db50fd681debddddcb555e838d2d9/sources/UI/UI/UIWindow.cs#L392) window procedure and then add to that a handler for `WM_PAINT` which *doesn't* use `BeginPaint`, `EndPaint`, but calls into the rendering instead. You can see how it is done in C++ in the DirectX samples [here](https://github.com/microsoft/DirectX-Graphics-Samples/blob/1704d8883ea74a73d332516bb6d990a134abaebd/Samples/Desktop/D3D12Fullscreen/src/Win32Application.cpp#L226).

The way to get `WndProc` to be called for a specific window instance is by forwarding like it is done [here](https://github.com/terrafx/terrafx/blob/3e3fa918543db50fd681debddddcb555e838d2d9/sources/UI/UI/UIService.cs#L144). Notice the `[UnmanagedCallersOnly]` and the use in there of `CREATESTRUCTW` to carry a `GCHandle` which targets the Window itself.